services/core-api/app/api/services/document_manager_service.py

```diff
@@ -19,6 +20,28 @@
 
 class DocumentManagerService():
     document_manager_document_resource_url = f'{Config.DOCUMENT_MANAGER_URL}/documents'
+    
+    @classmethod
+    def validateFileNameAndInitializeFileUploadWithDocumentManager(cls, request, mine, project_guid, document_category):
+
+        metadata = cls._parse_request_metadata(request)
+        file_name = metadata.get('filename')
+        mine_document = MineDocumentSearchUtil.find_by_document_name_and_project_guid(file_name, project_guid)
+
+        if not mine_document: # No existing file found in this application hence continuing the file uploading
```

Comment:
Can we add a "feature flag" to this so this won't block a release to prod until we're ready to go live with it?

Think elsewhere on the backend this has been done with:
`if Config.ENVIRONMENT_NAME != 'prod':`

Reply:
Added the flag to the [starting point](https://github.com/bcgov/mds/blob/mds-5273-minsespace-file-name-warning/services/core-api/app/api/projects/project_summary/resources/project_summary_document_upload.py#L29) of the new flow. And if its not PROD it will continue the same old flow.